src/optics/focusAtom.ts

```diff
@@ -35,6 +39,11 @@ export function focusAtom<S, A>(
   | WritableAtom<A | undefined, SetStateAction<A>>
   | WritableAtom<Array<A>, SetStateAction<A>>
   | PrimitiveAtom<S> {
+  const deps: object[] = [baseAtom, callback]
+  const cachedAtom: any = getWeakCacheItem(focusAtomCache, deps)
```

Comment:
Can we improve the typings here?

Reply:
Please help. The issue is more in `any` is L48. I spent some time and didn't find a good way.